downstairs/src/main.rs

```diff
@@ -1707,117 +1782,170 @@ async fn main() -> Result<()> {
                     .expect("Error init tracing subscriber");
             }
 
-            if let Some(oximeter) = oximeter {
-                let dssw = d.lock().await;
-                let dss = dssw.dss.clone();
+            // from https://docs.rs/slog/latest/slog/ - terminal out
+            let decorator = slog_term::TermDecorator::new().build();
+            let drain = slog_term::FullFormat::new(decorator).build().fuse();
+            let drain = slog_async::Async::new(drain).build().fuse();
 
-                tokio::spawn(async move {
-                    let new_address = match address {
-                        IpAddr::V4(ipv4) => {
-                            SocketAddr::new(std::net::IpAddr::V4(ipv4), 0)
-                        }
-                        IpAddr::V6(ipv6) => {
-                            SocketAddr::new(std::net::IpAddr::V6(ipv6), 0)
-                        }
-                    };
+            let log = slog::Logger::root(drain, slog::o!());
 
-                    if let Err(e) =
-                        stats::ox_stats(dss, oximeter, new_address).await
-                    {
-                        println!("ERROR: oximeter failed: {:?}", e);
-                    } else {
-                        println!("OK: oximeter all done");
-                    }
-                });
-            }
+            run_dropshot(bind_addr, &log).await
+        }
+    }
+}
+
+fn create_region(
+    block_size: u64,
+    data: PathBuf,
+    extent_size: u64,
+    extent_count: u64,
+    uuid: Uuid,
+    encrypted: bool,
+) -> Result<Region> {
+    /*
+     * Create the region options, then the region.
+     */
+    let mut region_options: crucible_common::RegionOptions = Default::default();
+    region_options.set_block_size(block_size);
+    region_options
+        .set_extent_size(Block::new(extent_size, block_size.trailing_zeros()));
+    region_options.set_uuid(uuid);
+    region_options.set_encrypted(encrypted);
+
+    let mut region = Region::create(&data, region_options)?;
+    region.extend(extent_count as u32)?;
+
+    Ok(region)
+}
+
+fn create_downstairs(
```

Comment:
We should probably name this something a little more descriptive.   We are creating the downstairs object, but the regions and data that make up that are already.  We have a create_region() which does indeed create a region.  Maybe `assemble_downstairs()` or `build_downstairs()` even?  Open to other suggestions as well.  I just thought a casual reader (like that will ever happen) might think a new downstairs is being created, but that is not exactly what this function does.